lms/djangoapps/ccx/tests/test_tasks.py

```diff
@@ -0,0 +1,102 @@
+"""
+Tests for celery tasks defined in tasks module
+"""
+
+from mock_django import mock_signal_receiver
+
+from ccx.tests.factories import (  # pylint: disable=import-error
+    CcxFactory,
+)
+from student.roles import CourseCcxCoachRole  # pylint: disable=import-error
+from student.tests.factories import (  # pylint: disable=import-error
+    AdminFactory,
+)
+from xmodule.modulestore.django import SignalHandler
+from xmodule.modulestore.tests.factories import CourseFactory
+from xmodule.modulestore.tests.django_utils import (
+    ModuleStoreTestCase,
+    TEST_DATA_SPLIT_MODULESTORE)
+from openedx.core.djangoapps.content.course_structures.models import CourseStructure
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from ccx_keys.locator import CCXLocator
+
+
+class TestSendCCXCoursePublished(ModuleStoreTestCase):
+    """unit tests for the send ccx course published task
+    """
+
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(TestSendCCXCoursePublished, self).setUp()
+        course = self.course = CourseFactory.create(org="edX", course="999", display_name="Run 666")
+        coach = AdminFactory.create()
+        role = CourseCcxCoachRole(course.id)
+        role.add_users(coach)
+        self.ccx = CcxFactory(course_id=course.id, coach=coach)
+
+    def call_fut(self, course_key):
+        """Call the function under test
+        """
+        from ..tasks import send_ccx_course_published
+        send_ccx_course_published(unicode(course_key))
+
+    def test_signal_not_sent_for_ccx(self):
+        """Check that course published signal is not sent when course key is for a ccx
+        """
+        course_key = CCXLocator.from_course_locator(self.course.id, self.ccx.id)
+        with mock_signal_receiver(SignalHandler.course_published) as receiver:
+            self.call_fut(course_key)
+            self.assertEqual(receiver.call_count, 0)
+
+    def test_signal_sent_for_ccx(self):
+        """Check that course published signal is sent when course key is not for a ccx
+        """
+        with mock_signal_receiver(SignalHandler.course_published) as receiver:
+            self.call_fut(self.course.id)
+            self.assertEqual(receiver.call_count, 1)
```

Comment:
Is it worth validating that this receiver sends out signals to *all* ccxs (rather than just one)?